llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -1192,3 +1207,66 @@ void transform_ext::makeSoftmaxMatcher(
       transform_ext::m_StructuredOp_Or(matcherContext, mulOperand, *divOperand);
   softmaxRootCapture = &softmaxRoot;
 }
+
+/// Matcher for convolutions.
+void transform_ext::makeConvolutionMatcher(
+    transform_ext::MatcherContext &matcherContext,
+    transform_ext::StructuredOpMatcher *&convolutionCapture,
+    transform_ext::StructuredOpMatcher *&fillCapture,
+    transform_ext::StructuredOpMatcher *&trailingCapture,
+    MatchedConvolutionCaptures &captures) {
+  // The core part of the matcher is anchored on a particular convolution op.
+  auto &nchwConvolution =
+      m_StructuredOp<linalg::Conv2DNchwFchwOp>(matcherContext);
+  auto &nhwcConvolution =
+      m_StructuredOp<linalg::Conv2DNhwcHwcfOp>(matcherContext);
```

Comment:
I believe it's possible to do `m_StrucutredOp<linalg::Conv2DNchwFchwOp, linalg::Conv2DNhwcHwcfOp>` and avoid the `OR`.